src/hotspots/HotspotTransformValues.js

```diff
@@ -0,0 +1,247 @@
+/**
+ * HotspotTransformValues handle the three values x, y and z.
+ *
+ * @constructor FORGE.HotspotTransformValues
+ * @param {number} x - The spherical coordinates of a 3D object (radius, theta, phi)
+ * @param {number} y - The rotation of a 3D object (x, y, z).
+ * @param {number} z - The scale of a 3D object (x, y, z).
+ * @extends {FORGE.BaseObject}
+ */
+FORGE.HotspotTransformValues = function(transformUid, x, y, z)
+{
+    /**
+     * The transformation related to this values.
+     * @name FORGE.HotspotTransformValues#_transformUid
+     * @type {string}
+     * @private
+     */
+    this._transformUid = transformUid;
+
+    /**
+     * The x.
+     * @name FORGE.HotspotTransformValues#_x
+     * @type {number}
+     * @private
+     */
+    this._x = typeof(x) === "number" ? x : 0;
+
+    /**
+     * The y.
+     * @name FORGE.HotspotTransformValues#_y
+     * @type {number}
+     * @private
+     */
+    this._y = typeof(y) === "number" ? y : 0;
+
+    /**
+     * The z.
+     * @name FORGE.HotspotTransformValues#_z
+     * @type {number}
+     * @private
+     */
+    this._z = typeof(z) === "number" ? z : 0;
+
+    /**
+     * onChange event dispatcher for transform values change.
+     * @name  FORGE.HotspotTransformValues#_onChange
+     * @type {FORGE.EventDispatcher}
+     * @private
+     */
+    this._onChange = null;
+
+    FORGE.BaseObject.call(this, "HotspotTransform");
```

Comment:
I don't think there should be an inheritance for a basic object like. Functions from `BaseObject` are not even used (as far as I have noted), so I suggest to remove it. It will reduce a little the memory usage and speed up a little things for this object which is used a lot.